src/Class/To.agda

```diff
@@ -0,0 +1,107 @@
+{-# OPTIONS --safe #-}
+module Class.To where
+
+record To (X : Set) (Y : Set) : Set where
+  field 
+    ⟦_⟧ : X → Y
+  infix 5 ⟦_⟧
+
+open To ⦃...⦄ public
+
+-- Metaprogram to derive instances
+open import Data.List.NonEmpty as NE using (List⁺; _∷_)
+open import Data.List
+open import Meta
+open import MetaPrelude
+open import Reflection.Tactic
+open import Reflection.Utils.Debug
+
+open import Class.MonadTC
+open import Class.Monad
+open import Class.Functor
+open import Class.Traversable
+
+open import Data.Product.Nary.NonDependent using (uncurryₙ)
+
+open MonadTCI
+
+getCodPi : Type → Type
+getCodPi ty =
```

Comment:
This function isn't really what you want here, since it strips away all the arguments of the field projections, but you want to only remove the first. Here's an example where deriving fails because of this:
```
record RFail : Set where
  field r : Bool → Bool

unquoteDecl To-RFail = derive-To ⦃ defaultTCOptions ⦄ ((quote RFail , To-RFail) ∷ [])
```
It's probably not an issue in this code base, but I could see this as a neat feature for stdlib-meta (Lean also has something equivalent to this, which is used everywhere).